keps/sig-network/2059-servicecidrblock-migration/README.md

```diff
@@ -0,0 +1,609 @@
+
+For enhancements that make changes to code or processes/procedures in core
+Kubernetes—i.e., [kubernetes/kubernetes], we require the following Release
+Signoff checklist to be completed.
+
+Check these off as they are completed for the Release Team to track. These
+checklist items _must_ be updated for the enhancement to be released.
+-->
+
+Items marked with (R) are required *prior to targeting to a milestone / release*.
+
+- [ ] (R) Enhancement issue in release milestone, which links to KEP dir in [kubernetes/enhancements] (not the initial KEP PR)
+- [ ] (R) KEP approvers have approved the KEP status as `implementable`
+- [ ] (R) Design details are appropriately documented
+- [ ] (R) Test plan is in place, giving consideration to SIG Architecture and SIG Testing input
+- [ ] (R) Graduation criteria is in place
+- [ ] (R) Production readiness review completed
+- [ ] Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+Provide a way to user to migrate a cluster to a different service CIDR Block range.
+
+## Motivation
+
+If user needs to change Service CIDR block, there should be a way to do it without 
+forcing them to recreate the cluster and migrate workload and/or causing downtime.
+
+I understand, that it might be not possible to automate it and provide OOTB support for it,
+but at least there should be a documented and approved way to do it. 
+This KEP is opened to get better understanding the complexity of doing it in existing clusters.
+
+<!--
+This section is for explicitly listing the motivation, goals and non-goals of
+this KEP.  Describe why the change is important and the benefits to users. The
+motivation section can optionally provide links to [experience reports] to
+demonstrate the interest in a KEP within the wider Kubernetes community.
+
+[experience reports]: https://github.com/golang/go/wiki/ExperienceReports
+-->
+
+### Goals
+
+1. There is a mechanism that can be used to help users migrate cluster to a different service CIDR block
+without causing downtime to the running workload.
+
+### Non-Goals
+
+
+## Proposal
+
+Proposed solution is the following:
+
+1. Allow kube-proxy to treat ExternalIPs from old Service CIDR block as ClusterIPs.
+2. Allow updating ClusterIP assigned to the Service. Allow user to specify `""` for the existing Service.
```

Comment:
@thockin clusterIP immutability again :-)

Reply:
The main reason for clusterIPs to be as draconian as they are is to make easier reconciliation against IaaS-ish systems.  We already have a gap with the ExternalName loophole.  It may be time to give up on this.  We have a richer ecosystem than we did "back then".

Perhaps the precursor KEP is "allow mutable ClusterIPs"